fossology/fossology
# feat(license): use spdx identifiers for reporting licenses

## Description
Add new field called as "SPDX ID" to all licenses allowing users/admins to add valid SPDX License Identifier to all licenses. This new ID will be used in all reports generated by FOSSology to get the license name.

In case, the field is empty, it will fallback to license's shortname, the current behavior.

![image](https://user-images.githubusercontent.com/18077542/208354047-493ded2f-1599-47a0-ad2f-f0d361ceaabc.png)

At the same time, update nomos to spit out new SPDX identifiers for licenses like `GPL-2.0` to `GPL-2.0-only`.

### Changes

1. Rename field `rf_spdx_compatible` to `rf_spdx_id` in `license_ref` table.
2. Use the new field for license name in reports dao.
3. New function `getProjectedSpdxId()` to `LicenseMap`, used to get license name in reports.
4. Fix CLIXML report.
5. Update license names in nomos to use new SPDX naming standards.

## How to test

1. Install FOSSology and scan some package.
2. Make sure FOSSology display's new SPDX names for licenses.
3. Update SPDX ID for some identified licenses and generate reports.
4. Check if the report contains updated SPDX ID for license name.
5. For licenses missing SPDX ID, the report should contain license's shortname.

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2363"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Closes #748
Closes #1061
Closes #1365
